src/common/helper.js

```diff
@@ -543,20 +552,26 @@ function * checkGetAccess (authUser, submission, parentSpan) {
         const screener = _.filter(currUserRoles, { role: 'Primary Screener' })
         const reviewer = _.filter(currUserRoles, { role: 'Reviewer' })
 
-        // User is either a Reviewer or Screener
-        if (screener.length !== 0 || reviewer.length !== 0) {
+        if (screener.length !== 0) { // User is Screener
           const screeningPhase = _.filter(phases, { phaseType: 'Screening', phaseStatus: 'Scheduled' })
+
+          // Screening is not Opened / Closed
+          if (screeningPhase.length !== 0) {
+            throw new errors.HttpStatusError(403, 'You can access the submission only when Screening is open')
+          }
+        } else if (reviewer.length !== 0) { // User is Reviewer
           const reviewPhase = _.filter(phases, { phaseType: 'Review', phaseStatus: 'Scheduled' })
 
-          // Neither Screening Nor Review is Opened / Closed
-          if (screeningPhase.length !== 0 && reviewPhase.length !== 0) {
-            throw new errors.HttpStatusError(403, 'You can access the submission only when Screening / Review is open')
+          // Review is not Opened / Closed
+          if (reviewPhase.length !== 0) {
+            throw new errors.HttpStatusError(403, 'You can access the submission only when Review is open')
           }
         } else {
           const appealsResponse = _.filter(phases, { phaseType: 'Appeals Response', 'phaseStatus': 'Closed' })
 
           // Appeals Response is not closed yet
-          if (appealsResponse.length === 0) {
+          // Do not check for end of Appeals Response for Design Challenges
+          if (appealsResponse.length === 0 && track !== 'DESIGN') {
```

Comment:
@sharathkumaranbu According to the [Lucidchart](https://www.lucidchart.com/documents/edit/d4f1f6ec-ddde-4046-a85e-ab2804af41c1/0_0?shared=true), a submitter can download other submissions of `DESIGN` challenges if they have at least 1 submission passing screening score.

Would the check `submission.reviewSummation[0].isPassing` be valid for checking whether the submission has passed screening? Or is this check only valid for checking passing reviews?

Reply:
Unsure about this. @rootelement Thoughts on the above? My understanding is that review summation is an aggregate of the reviews and in this scenario, we need to check for the review of type screening...